### PR TITLE
PA-4-create-workflow

### PR DESCRIPTION
**Pull request by seanchan-lcjg**
Create workflow for Club Monaco Persona:
1. Create datamart
2. generate tagging parquet
3. generate model feature
4. predict data by previous kmeans model
5. create profiling of the customer segment (SOW, no. of customer)
6. export csv (no. of customer in each cluster)
7. send email (SOW of main class, sub class, no. of customer)

### CODE REVIEW

File: workflow/workflow.py
Comment (cintiaching) [inline workflow/workflow.py:1]: restructure the files as discussed, `workflow` folder seems unnecessary

File: workflow/profiling/profile_and_export.py [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+dbutils.widgets.text("end_date", "") 
+dbutils.widgets.text("base_dir", "")
+
+# COMMAND ----------
+
+import os
+import pyspark.sql.functions as f
+
+datamart_dir = os.path.join(dbutils.widgets.get("base_dir"), "datamart")
+
```
Comment (cintiaching) [inline workflow/profiling/profile_and_export.py:13]: nice

File: workflow/profiling/profile_and_export.py [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+  'no. of customer' AS Persona,
+  SUM(CASE WHEN persona = 'Prime Connoisseur' THEN count END) AS Prime_Connoisseur,
+  SUM(CASE WHEN persona = 'Menswear Specialist' THEN count END) AS Menswear_Specialist,
+  SUM(CASE WHEN persona = 'Outerwear Fashionista' THEN count END) AS Outerwear_Fashionista,
+  SUM(CASE WHEN persona = 'Bottoms and Dresses Diva' THEN count END) AS Bottoms_and_Dresses_Diva,
+  SUM(CASE WHEN persona = 'Average Fashion Connoisseur' THEN count END) AS Average_Fashion_Connoisseur,
+  SUM(CASE WHEN persona IS NULL THEN count END) AS Total
+FROM persona_counts
+""").toPandas()
+
```
Comment (cintiaching) [inline workflow/profiling/profile_and_export.py:103]: as per Arnab's comments, the schema should be
```
vip_no
persona
date
```
and whatever he selected from olga's list

File: workflow/profiling/profile_and_export.py [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+    +f"<b>from {start_date} - {end_date}</b>"
+    +"<br><br><b>No. of customers:</b><br><br>"
+    + persona_df.to_html(index=False)
+    + "<br><br><b>Persona SOW by maincat:</b><br><br>"
+    + maincat_df.to_html(index=False)
+    + "<br><br><b>Persona SOW by subcat:</b><br><br>"
+    + subcat_df.to_html(index=False)
+)
+tracker_email_subject = f"Club Monaco - SOW in {quarter_year} Q{quarter_no}"
+send_email(
```
Comment (cintiaching) [inline workflow/profiling/profile_and_export.py:308]: rmb to change the email title as arnab mentioned, and add the csv as attachment

**Review by cintiaching — CHANGES_REQUESTED** [review]

File: model_features.py [hunk 4]
```diff
@@ -1,27 +1,33 @@
...
+dbutils.widgets.text("base_dir", "")
+
+# COMMAND ----------
+
 import os
 import pandas as pd
 import pyspark.sql.functions as f
 
-base_dir = "/mnt/dev/customer_segmentation/imx/club_monaco/datamart"
+datamart_dir = dbutils.widgets.get("base_dir") + "/datamart"
```
Comment (cintiaching) [inline model_features.py:13]: `datamart_dir = os.path.join(dbutils.widgets.get("base_dir"), "datamart")`

**Review by cintiaching — APPROVED** [review]
lgtm, thanksss

File: workflow/model_features.py [hunk 5]
```diff
@@ -1,27 +1,33 @@
...
+dbutils.widgets.text("base_dir", "")
+
+# COMMAND ----------
+
 import os
 import pandas as pd
 import pyspark.sql.functions as f
 
-base_dir = "/mnt/dev/customer_segmentation/imx/club_monaco/datamart"
+datamart_dir = dbutils.widgets.get("base_dir") + "/datamart"
```
[the remark this answers is not in the document]
Comment (seanchan-lcjg) [reply workflow/model_features.py]: changed

**Pull request merged**